src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/naxxramas/instance_naxxramas.cpp

```diff
@@ -527,21 +530,32 @@ void instance_naxxramas::SetData(uint32 type, uint32 data)
                 break;
             }
             if (data == FAIL)
+            {
                 m_horsemenKilled = 0;
+                {
+                    std::vector<uint32> entries = {NPC_BLAUMEUX, NPC_MOGRAINE, NPC_ZELIEK, NPC_THANE};
+                    for (uint32 entry : entries)
+                        if (Creature* creature = GetSingleCreatureFromStorage(entry))
+                            if (creature->IsInCombat())
+                                creature->AI()->EnterEvadeMode();
+                }
+            }
             m_auiEncounter[type] = data;
             DoUseDoorOrButton(GO_MILI_HORSEMEN_DOOR);
+
+            // Despawn spirits and void zones
+            {            
+                std::vector<uint32> entries = {NPC_SPIRIT_OF_BLAUMEUX, NPC_SPIRIT_OF_MOGRAINE, NPC_SPIRIT_OF_KORTHAZZ, NPC_SPIRIT_OF_ZELIREK};
+                for (uint32 entry : entries)
+                    if(Creature* spirit = GetSingleCreatureFromStorage(entry))
+                        spirit->ForcedDespawn();
+                GuidVector voidzones;
+                GetCreatureGuidVectorFromStorage(NPC_BLAUMEUX_VOIDZONE, voidzones);
```

Comment:
I know this is kinda out of scope for the pull request, but I noticed and my brain compelled me to fix it anyway. Should I keep it, remove it or put it into a separate pull request?